client/src/escrow_client/mod.rs

```diff
@@ -1,155 +1,152 @@
+    /// State change for the buyer. The state after that is token sent.
+    ///
+    /// Returns the sent trade token by this [`EscrowClient`].
+    async fn send_trade_token(&self, nostr_client: &NostrClient) -> anyhow::Result<String> {
+        let escrow_contract = &self.prev_state.escrow_contract;
+        let escrow_token = self
+            .prev_state
+            .ecash_wallet
+            .create_escrow_token(escrow_contract, &self.escrow_registration)
             .await?;
 
-        let escrow_coordinator_pk =
-            Self::receive_escrow_coordinator_pk(nostr_client, escrow_coordinator_npub).await?;
-        Ok(escrow_coordinator_pk)
-    }
+        debug!("Sending token to the seller: {}", escrow_token);
+
+        nostr_client
+            .client
+            .send_private_msg(escrow_contract.npubkey_seller, &escrow_token, None)
+            .await?;
+        dbg!("Sent Token to seller");
 
-    async fn parse_escrow_pk(pk_message_json: &String) -> anyhow::Result<(PublicKey, Timestamp)> {
-        let pkm: PubkeyMessage = serde_json::from_str(pk_message_json)?;
-        let public_key = PublicKey::from_hex(pkm.escrow_coordinator_pubkey)?;
-        Ok((public_key, pkm.escrow_start_ts))
+        Ok(escrow_token)
     }
 
-    async fn receive_escrow_coordinator_pk(
+    /// State change for a seller. The state after this is token received.
+    ///
+    /// Returns the received trade token by this [`EscrowClient`].
+    async fn receive_and_validate_trade_token(
+        &self,
         nostr_client: &NostrClient,
-        coordinator_npub: &String,
-    ) -> anyhow::Result<(PublicKey, Timestamp)> {
-        let filter_note = Filter::new()
-            .kind(Kind::EncryptedDirectMessage)
-            .since(Timestamp::now())
-            .author(nostr_sdk::PublicKey::from_bech32(coordinator_npub)?);
-        nostr_client.client.subscribe(vec![filter_note], None).await;
-
-        let mut notifications = nostr_client.client.notifications();
-
-        while let Ok(notification) = notifications.recv().await {
-            if let RelayPoolNotification::Event { event, .. } = notification {
-                if let Some(decrypted) = nostr_client
-                    .decrypt_msg(&event.content, &event.author())
-                    .await
-                {
-                    dbg!("Received event: {:?}", &decrypted);
-                    if let Ok(pk_ts) = Self::parse_escrow_pk(&decrypted).await {
-                        nostr_client.client.unsubscribe_all().await;
-                        return Ok(pk_ts);
-                    }
-                }
-            }
-        }
-        Err(anyhow!("No valid escrow coordinator public key received"))
+    ) -> anyhow::Result<Token> {
+        let escrow_contract = &self.prev_state.escrow_contract;
+        let wallet = &self.prev_state.ecash_wallet;
+
+        let message = nostr_client.receive_escrow_message(20).await?;
+        dbg!("Received Token, vaidating it...");
```

Comment:
We probably should use debug!/trace!/error!/warn! everywhere instead of mixing dbg!/println! and logging

Reply:
Yes, we need a unified procedure for this aspect, especially for a production mode. Let's handle this in a separate issue: https://github.com/f321x/cashu-escrow-kit/issues/13.